### DISCUSSION TITLE
Conversion from GWT

### DISCUSSION THREAD

**Discussion by spardue** [google/j2cl#159] [Q&A] [answered]
Hello,

Any chance there will be a tool to convert over from GWT to this project?

**Comment by niloc132** [comment, marked as answer]
If you are using JSNI, you can already translate that code to JsInterop, which works nearly the same in both projects. If you are using Generators, these can almost all be rewritten as Annotation Processors. As for the rest of the ecosystem, many of the components have been updated in this same way already, so that they are compatible with either GWT2 or J2CL projects.

The largest issue for most projects is likely to be the changes in how the compiler is meant to be part of some other build system rather than run its own de facto build tool, where it will behave very similarly to `javac`, etc. The biggest impact to Maven-based GWT projects will be encouraging/requiring them to break up client/server/shared into distinct modules, rather than one big classpath to hold everything.

**Comment by treblereel** [comment]
In my company I am responsible for gwt to j2cl migration/tooling and I successfully finished several projects, including a huge one (like 80 maven modules and tons of external deps).

So it's achievable, but you have to perform some work by yourself. What is nice, there are many projects that are ported/adopted to work with gwt and j2cl.  The main problem is gwt-user modules. We (as a community) have ported most of them to j2cl, but few are left. We still haven't finished i18n, which can be a stopper for you (well, i have my own port and it works somehow, but @niloc312 thinks, we can do better :) ).



We are mostly focused on maven as a build tool and the latest j2cl-maven-plugin is pretty stable, so you can check my articles on [dev.to](https://dev.to/treblereel) how to start with it. Feel free to ping me in gitter if you have any question.
